Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UnitData.xml

```diff
@@ -7699,11 +7699,31 @@
         <AbilArray Link="AP_MorphlingToViper"/>
         <AbilArray Link="AP_MorphlingToGuardianSCBW@Morph"/>
         <AbilArray Link="AP_MorphlingToDevourerSCBW@Morph"/>
+        <AbilArray Link="AP_MorphlingToDevourerSCBW@Morph"/>
+        <AbilArray Link="AP_MorphlingToPrimalIgniterMerge"/>
         <CardLayouts>
             <LayoutButtons Face="Move" Type="AbilCmd" AbilCmd="move,Move" Row="0" Column="0"/>
             <LayoutButtons Face="Stop" Type="AbilCmd" AbilCmd="stop,Stop" Row="0" Column="1"/>
             <LayoutButtons Face="MoveHoldPosition" Type="AbilCmd" AbilCmd="move,HoldPos" Row="0" Column="2"/>
             <LayoutButtons Face="MovePatrol" Type="AbilCmd" AbilCmd="move,Patrol" Row="0" Column="3"/>
+            <LayoutButtons>
+                <Face value="AP_MorphlingSwarmMorphs"/>
+                <Type value="Submenu"/>
+                <AbilCmd value="255"/>
+                <SubmenuCardId value="MSw"/>
+                <Row value="2"/>
+                <Column value="0"/>
+            </LayoutButtons>
+            <LayoutButtons>
+                <Face value="AP_MorphlingPrimalMorphs"/>
+                <Type value="Submenu"/>
+                <AbilCmd value="255"/>
+                <SubmenuCardId value="MPr"/>
+                <Row value="2"/>
+                <Column value="1"/>
+            </LayoutButtons>
+        </CardLayouts>
+        <CardLayouts CardId="MSw">

```

Comment:
I think there's some kind of flag you can put on the `CardLayouts` tags (something like `sticky="true"`?) that makes the submenus "sticky". This means that if you select an ability from the submenu, the submenu will stay open for you to click it again. I think that's desirable here in the case where you want to morph / merge multiple morphlings to the same thing.